src/analytics/utils/analytics_mapping_utils.py

```diff
@@ -0,0 +1,205 @@
+# Analytics related mapping functions. Needed for sending structured data to analytics providers such as Segment.io and Amazon Personalize.
+
+import time
+from datetime import datetime
+
+import segment.analytics as analytics
+
+from discussion.reaction_models import Vote
+from paper.utils import format_raw_authors
+from researchhub_comment.related_models.rh_comment_model import RhCommentModel
+
+
+def log_analytics_event(action):
```

Comment:
should tipping and contributing to a fundraise technically be tracked as user interactions too? perhaps it's something we have to implement in the future.

Reply:
Yeah, as a next step I am going to add more events such as tipping